account_tax_expense_include/account_tax.py

```diff
@@ -30,3 +30,92 @@ class account_tax(orm.Model):
+        context = context or {}
+
+        res = super(account_tax, self).write(
+            cr, uid, ids, vals, context=context)
+
+        for tax in self.browse(cr, uid, ids, context=context):
+            for child in tax.child_ids:
+                child.write({'expense_include': tax.expense_include})
+
+        return res
+
+    def compute_all(self, cr, uid, taxes, price_unit, quantity, product=None,
+                    partner=None, force_excluded=False, is_expense=False):
+        """
+        :param force_excluded: boolean used to say that we don't want to
+            consider the value of field price_include of tax. It's used in
+            encoding by line where you don't matter if you encoded a tax with
+            that boolean to True or False
+        :param is_expense: boolean used to say that we're calculating taxes for
+            expense move lines
+        RETURN: {
+                'total': 0.0,         # Total without taxes
+                'total_included: 0.0, # Total with taxes
+                'taxes': []           # Taxes, see compute for the format
+            }
+        """
+
+        if not is_expense:
+            return super(account_tax, self).compute_all(
+                cr, uid, taxes, price_unit, quantity,
+                product=product, partner=partner, force_excluded=force_excluded
+            )
+
+        # By default, for each tax, tax amount will first be computed
+        # and rounded at the 'Account' decimal precision for each
+        # PO/SO/invoice line and then these rounded amounts will be
+        # summed, leading to the total amount for that tax. But, if the
+        # company has tax_calculation_rounding_method = round_globally,
+        # we still follow the same method, but we use a much larger
+        # precision when we round the tax amount for each line (we use
+        # the 'Account' decimal precision + 5), and that way it's like
+        # rounding after the sum of the tax amounts of each line
+        precision = self.pool.get('decimal.precision').precision_get(
+            cr, uid, 'Account')
+        tax_compute_precision = precision
+        if taxes and taxes[0].company_id.\
+                tax_calculation_rounding_method == 'round_globally':
+            tax_compute_precision += 5
+        totalin = totalex = round(price_unit * quantity, precision)
+        tin = []
+        tex = []
+        for tax in taxes:
+            if (tax.expense_include or tax.price_include)\
+                    and not force_excluded:
+                tin.append(tax)
+            else:
+                tex.append(tax)
+
+        tin = self.compute_inv(
+            cr, uid, tin, price_unit, quantity, product=product,
+            partner=partner, precision=tax_compute_precision)
+        for r in tin:
+            totalex -= r.get('amount', 0.0)
+        totlex_qty = 0.0
+        try:
+            totlex_qty = totalex / quantity
+        except:
+            pass
```

Comment:
Please explain why you're swallowing all errors. It's better to specify the errors you're expecting to get.